### PR TITLE
add CI for podman_crun

### PR DESCRIPTION
**Pull request by JcJinChen**
Add the CI for podman + crun

### CODE REVIEW

File: podman_crun/README.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+```bash
+git clone https://github.com/containers/crun
+cd crun
+./autogen.sh
+./configure --with-wasmedge
+make
+sudo make install
+```
+
+## Install containerd
```
Comment (juntao) [inline podman_crun/README.md:57]: Do we need containerd here?

File: podman_crun/README.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+## Run a simple WebAssembly app
+
+Finally, we can run a simple WebAssembly program using containerd.
+[A seperate article](../simple_wasi_app.md) explains how to compile, package, and publish the WebAssembly
+program as a container image to Docker hub.
+In this section, we will start off pulling this WebAssembly-based container
+image from Docker hub using podman.
+
+```bash
+sudo podman pull docker.io/zjujinchen/mywasm-image:latest
```
Comment (juntao) [inline podman_crun/README.md]: We should use the same image for all demos.

File: podman_crun/simple_wasi_application.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+sudo podman pull docker.io/zjujinchen/mywasm-image
```
Comment (juntao) [inline podman_crun/simple_wasi_application.sh]: Use the same image as other workflows.

File: podman_crun/install.sh [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+
+wget https://github.com/containerd/containerd/releases/download/v${VERSION}/cri-containerd-cni-${VERSION}-linux-amd64.tar.gz
+wget https://github.com/containerd/containerd/releases/download/v${VERSION}/cri-containerd-cni-${VERSION}-linux-amd64.tar.gz.sha256sum
+sha256sum --check cri-containerd-cni-${VERSION}-linux-amd64.tar.gz.sha256sum
+
+sudo tar --no-overwrite-dir -C / -xzf cri-containerd-cni-${VERSION}-linux-amd64.tar.gz
+sudo systemctl daemon-reload
+
+# change containerd conf to use crun as default
```
Comment (juntao) [inline podman_crun/install.sh]: Do we need containerd?

**Review by juntao — CHANGES_REQUESTED** [review]

File: podman_crun/README.md (hunk 2 above)
Comment (JcJinChen) [reply podman_crun/README.md]: Actually, this image that I builded is familiar for me. Of course, I just try the image builded by hydai. It works as well, I will use the same images and commits again.

File: podman_crun/README.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+## Run a simple WebAssembly app
+
+Finally, we can run a simple WebAssembly program using containerd.
+[A seperate article](../simple_wasi_app.md) explains how to compile, package, and publish the WebAssembly
+program as a container image to Docker hub.
+In this section, we will start off pulling this WebAssembly-based container
+image from Docker hub using podman.
+
+```bash
+sudo podman pull docker.io/hydai/wasm-wasi-example:with-wasm-annotation
```
Comment (juntao) [inline podman_crun/README.md:93]: Do we still use hydai's image here?

File: podman_crun/README.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+```
+
+We can run the example in just one line with podman
+
+```bash
+sudo podman run docker.io/hydai/wasm-wasi-example:with-wasm-annotation  /wasi_example_main.wasm 50000000
```
Comment (juntao) [inline podman_crun/README.md:99]: Use hydai's image here?

File: podman_crun/simple_wasi_application.sh [hunk 7]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+sudo podman pull docker.io/hydai/wasm-wasi-example:with-wasm-annotation
```
Comment (juntao) [inline podman_crun/simple_wasi_application.sh:2]: Hydai's image?

File: simple_wasi_app.md [hunk 8]
```diff
@@ -40,15 +41,15 @@ ADD wasi_example_main.wasm /
 CMD ["/wasi_example_main.wasm"]
 ```
 
-## Create container image with annotations
+### Create container image with annotations
 
 > Please note that adding self-defined annotation is still a new feature in buildah.
 
 The `crun` container runtime can start the above WebAssembly-based container image. But it requires the `module.wasm.image/variant=compat` annotation on the container image to indicate that it is a WebAssembly application without a guest OS. You can find the details in [Official crun repo](https://github.com/containers/crun/blob/main/docs/wasm-wasi-example.md).
```
Comment (juntao) [inline simple_wasi_app.md:48]: Should we use the smart annotation here?

File: simple_wasi_app.md [hunk 9]
```diff
@@ -40,15 +41,15 @@ ADD wasi_example_main.wasm /
...
 
 To add `module.wasm.image/variant=compat` annotation in the container image, you will need the latest [buildah](https://buildah.io/). Currently, Docker does not support this feature. Please follow [the install instructions of buildah](https://github.com/containers/buildah/blob/main/install.md) to build the latest buildah binary.
```
Comment (juntao) [inline simple_wasi_app.md:50]: Use smart annotation?

**Comment by juntao** [comment]
flows summarize

**Comment by juntao** [comment]
Hello, I am a [serverless review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/). Here are my reviews of code commits in this PR.

------

**Comment by hydai** [comment]
This is outdated. Closing it. Thanks.

**Pull request closed without merging**